ibis/backends/flink/__init__.py

```diff
@@ -315,30 +343,34 @@ def create_table(
         import pyarrow_hotfix  # noqa: F401
 
         import ibis.expr.types as ir
+        from ibis.backends.flink.datatypes import FlinkRowSchema
 
         if obj is None and schema is None:
             raise exc.IbisError("The schema or obj parameter is required")
 
-        if overwrite:
-            self.drop_table(name=name, catalog=catalog, database=database, force=True)
+        # in-memory data is created as views in `pyflink`
+        elif obj is not None:
+            if isinstance(obj, pa.Table):
+                obj = obj.to_pandas()
+            if isinstance(obj, pd.DataFrame):
+                table = self._table_env.from_pandas(
+                    obj, FlinkRowSchema.from_ibis(schema)
+                )
+            if isinstance(obj, ir.Table):
+                table = obj
+            return self.create_view(name, table, database=database, overwrite=overwrite)
 
-        if isinstance(obj, pa.Table):
-            obj = obj.to_pandas()
-        if isinstance(obj, pd.DataFrame):
-            qualified_name = self._fully_qualified_name(name, database, catalog)
-            table = self._table_env.from_pandas(obj)
-            # in-memory data is created as views in `pyflink`
-            # TODO(chloeh13q): alternatively, we can do CREATE TABLE and then INSERT
-            # INTO ... VALUES to keep things consistent
-            self._table_env.create_temporary_view(qualified_name, table)
-        if isinstance(obj, ir.Table):
-            # TODO(chloeh13q): implement CREATE TABLE for expressions
-            raise NotImplementedError
-        if schema is not None:
+        # external data is created as tables in `pyflink`
```

Comment:
Is another way of thinking of "external data" as data that requires a "connector"?  Asking for my own understanding.

Reply:
Yes, your understanding is correct. So "external tables" require specific connector configurations (captured by `tbl_properties`) for Flink to connect to as either sources or sinks. 

Food for thought: it may require a deeper discussion, but how does Ibis think about reusable connectors? I'd love to see a world where connectors can implemented once and reused cross different engines or even runtimes (e.g., leverage arrow or in memory data sharing protocol). Today, the data ecosystem is plagued with connectors, every single vendor reimplements the same connectors over and over again. This may not be within the scope of Ibis, but wanted to throw the idea out there.

Reply:
We like reusable things and avoiding many-to-many style connectors -- I'm not sure if Ibis has a role to play here beyond adoption and cheerleading.